homework/index.js

```diff
@@ -30,15 +32,119 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-        return;
-      }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+  function createOptions(repositories, select) {
+    repositories.forEach((repository, index) => {
+      createAndAppend('option', select, {
+        text: repository.name,
+        value: index,
+      });
+    });
+  }
+
+  function addRow(tbody, label, value) {
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('th', tr, {
+      text: `${label.toUpperCase()}: `,
+      class: label,
+    });
+    createAndAppend(
+      'td',
+      tr,
+      value
+        ? {
+            text: value,
+          }
+        : undefined,
```

Comment:
There is a slight issue here. I would like to know if a repo has zero forks. However, `0` is also falsy. Therefore the number zero will not be displayed with your code. This will work:

```js
createAndAppend('td', tr, value == null ? undefined : { text: value });
```

`value == null` with be true if value is `null` or `undefined`. Comparing to `null` like this is the only time you can use a non-strict comparison with `==`.

Reply:
Thank you for your feedback, I have just tried it and solved it.